lib/git.ts

```diff
@@ -21,6 +21,9 @@ function trimTrailingNewline(str: string): string {
 }
 
 export function git(args: string[], options?: IGitOptions): Promise<string> {
+    // allow the command to run in a bare repository
+    if (options?.workDir?.endsWith(".git")) args = [`--git-dir=${options.workDir}`, ...args];
```

Comment:
Noting the `.git` suffix is not a git requirement but just a convention and maybe a best practice.  This would make it a requirement for users of ggg.  Not a big deal but would need to be documented.

Sorry if I missed doc that indicated it was a git requirement.

Reply:
> Noting the `.git` suffix is not a git requirement but just a convention and maybe a best practice.

It is also what `git clone --bare <url>` appends to the newly-created directory's name.

> This would make it a requirement for users of ggg.  Not a big deal but would need to be documented.

Yup, I'll document it.

> Sorry if I missed doc that indicated it was a git requirement.

I should also have been clearer that the intention here is to prepare for [`safe.bareRepository = explicit`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-safebareRepository) (which should have been made the default a long time ago but the Git project drags feet about it).

Reply:
> > This would make it a requirement for users of ggg.  Not a big deal but would need to be documented.
> 
> Yup, I'll document it.

Actually, this only affects GitGitGadget's _own_ use. So I think we'll be fine.

Reply:
> Actually, this only affects GitGitGadget's _own_ use. So I think we'll be fine.

Wouldn't this affect anyone trying to use a bare repo?

Reply:
> Wouldn't this affect anyone trying to use a bare repo?

By anyone I meant anyone using GitGitGadget to submit changes for repos other than git (by providing their own config).  Wouldn't these users also want to use bare repos in their workflows?

Reply:
> By anyone I meant anyone using GitGitGadget to submit changes for repos other than git (by providing their own config). Wouldn't these users also want to use bare repos in their workflows?

The idea is that the Azure Pipelines (and in the future, GitHub workflows) will initialize their local worktree (or bare repository) themselves, i.e. they have full control over the naming.

In any case, in the future, I think, we should _only_ use bare repositories, read: eventually this logic will go away and the callers will have to specify the `<worktree>/.git` path _iff_ they still want to work on non-bare rather than bare repositories.